### PR TITLE
feat(github-growth): fetch missing members in API

### PR DESCRIPTION
**Pull request by cathteng**
Get the GitHub commit authors that are not members of the organization for commits made in the last 30 days (first pass)

For ER-1753 and ER-1754

### CODE REVIEW

File: tests/sentry/api/test_organization_missing_members.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        self.repo = self.create_repo(project=self.project, provider="integrations:github")
+        self.create_commit(repo=self.repo, author=self.commit_author)
+        self.create_commit(repo=self.repo, author=self.commit_author1)
+        self.create_commit(repo=self.repo, author=self.commit_author1)
+        self.create_commit(repo=self.repo, author=self.commit_author2)
+
+        self.login_as(self.user)
+
+    def test_simple(self):
```
Comment (AniketDas-Tekky) [inline tests/sentry/api/test_organization_missing_members.py:38]: Needs some more tests like what if there's a non-github commit author, or what if there are no commit authors?

Comment (cathteng) [reply tests/sentry/api/test_organization_missing_members.py:42]: added more

File: src/sentry/api/endpoints/organization_missing_org_members.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    permission_classes = (MissingMembersPermission,)
+
+    def _get_missing_members(self, organization: Organization) -> QuerySet[CommitAuthor]:
+        member_emails = set(
+            organization.member_set.exclude(email=None).values_list("email", flat=True)
+        )
+        member_emails.update(
+            set(
+                organization.member_set.exclude(user_email=None).values_list(
+                    "user_email", flat=True
```
Comment (EricHasegawa) [inline src/sentry/api/endpoints/organization_missing_org_members.py:37]: What's the difference between `email` and `user_email`?

File: src/sentry/api/endpoints/organization_missing_org_members.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                )
+            )
+        )
+        nonmember_authors = CommitAuthor.objects.filter(organization_id=organization.id).exclude(
+            email__in=member_emails
+        )
+
+        # This is currently for Github only
```
Comment (EricHasegawa) [inline src/sentry/api/endpoints/organization_missing_org_members.py:45]: Will this ever be for anything other than GitHub? If so, who/what would this be for - other version control systems?

File: src/sentry/api/endpoints/organization_missing_org_members.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        return (
+            nonmember_authors.filter(
+                commit__repository_id__in=set(
```
Comment (EricHasegawa) [inline src/sentry/api/endpoints/organization_missing_org_members.py:48]: Can we filter this out into some intermediate variables to make it more obvious what's happening?

File: src/sentry/api/endpoints/organization_missing_org_members.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                commit__date_added__gte=timezone.now() - timedelta(days=30),
+            )
+            .annotate(commit_count=Count("commit"))
+            .order_by("-commit_count")
+        )
+
+    # TODO(cathy): check domain
+
+    def get(self, request: Request, organization) -> Response:
+        # TODO(cathy): search
```
Comment (EricHasegawa) [inline src/sentry/api/endpoints/organization_missing_org_members.py:62]: What will search look like?

File: src/sentry/api/endpoints/organization_missing_org_members.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                        provider="integrations:github", organization_id=organization.id
+                    ).values_list("id", flat=True)
+                ),
+                commit__date_added__gte=timezone.now() - timedelta(days=30),
+            )
+            .annotate(commit_count=Count("commit"))
+            .order_by("-commit_count")
+        )
+
+    # TODO(cathy): check domain
```
Comment (EricHasegawa) [inline src/sentry/api/endpoints/organization_missing_org_members.py:59]: what does this entail and what will it do for us

File: tests/sentry/api/test_organization_missing_members.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        self.create_member(
+            email="a@example.com",
+            organization=self.organization,
+        )
+        member = self.create_member(user=self.create_user(), organization=self.organization)
+        member.user_email = "b@example.com"
+        member.save()
+
+        self.commit_author = self.create_commit_author(project=self.project, email="b@example.com")
+        self.commit_author1 = self.create_commit_author(project=self.project, email="c@example.com")
```
Comment (EricHasegawa) [inline tests/sentry/api/test_organization_missing_members.py:26]: Could we give these better names so we know what's different about them? Like if one is not a member name them `non_member_commit_author` or whatever

**Review by EricHasegawa — COMMENTED** [review]
Mostly just questions so I can gain context - looks great :) Will approve once I understand better

File: src/sentry/api/endpoints/organization_missing_org_members.py (hunk 3 above)
Comment (cathteng) [reply src/sentry/api/endpoints/organization_missing_org_members.py:45]: yes other version control systems. next logical one would be gitlab

File: src/sentry/api/endpoints/organization_missing_org_members.py (hunk 5 above)
Comment (cathteng) [reply src/sentry/api/endpoints/organization_missing_org_members.py:62]: this is to query for members with certain emails like in a search bar

File: src/sentry/api/endpoints/organization_missing_org_members.py (hunk 6 above)
Comment (cathteng) [reply src/sentry/api/endpoints/organization_missing_org_members.py:59]: sometimes users in an organization will have different domains in their email addresses and we want to filter for ones that match the owners' email domains to be sure that the users actually belong to the organization

File: src/sentry/api/endpoints/organization_missing_org_members.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    def get(self, request: Request, organization) -> Response:
+        # TODO(cathy): search
+        queryset = self._get_missing_members(organization)
+
+        return Response(
```
Comment (schew2381) [inline src/sentry/api/endpoints/organization_missing_org_members.py:65]: Does this need to be returned as a list? We could add the integration as a query param to only return a specific integration instead

**Review by AniketDas-Tekky — APPROVED** [review]

**Pull request merged**